File: gabriel/Problema0350/Problema0350.java
package Problema0350;

public class Problema0350 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
         
          System.out.println("Digite uma palavra");
          String palavra = input.nextLine();
          
          System.out.println("Digite outra palavra");
          String palavra2 = input.nextLine();
          
          char vetorPalavra[] = palavra.toCharArray();
          char vetorPalavra2[] = palavra2.toCharArray();
          boolean permuta = true;
          
          if(palavra.length() != palavra2.length()){
              permuta = false;
          }
          
          else{
          
              for(int i=0; i<palavra.length(); i++){
                  for(int j=0; j<palavra2.length(); j++){
                      if(vetorPalavra[i] == vetorPalavra2[j]){
                          vetorPalavra[i] = 0;
                          vetorPalavra2[j] = 0;
                      }
                  }
              }
          
              for(int i=0; i<palavra.length(); i++){
                  if(vetorPalavra[i] != 0 || vetorPalavra2[i] != 0){
                      permuta = false;
                  }
              }
          }   
          System.out.println("A primeira palavra é uma permutação da segunda? : " +permuta);
       
   }
  }

File: gabriel/Problema0350/Problema0350.py
palavra = input("Digite uma palavra: ")
palavra2 = input("Digite outra palavra: ")

vetorPalavra = [0] * len(palavra)
vetorPalavra2 = [0] * len(palavra2)
permuta = True

for i in range(len(palavra)):
  vetorPalavra[i] = palavra[i]

for i in range(len(palavra2)):
  vetorPalavra2[i] = palavra2[i]

if(len(palavra) != len(palavra2)):
  permuta = False
        
        
else:
        
  for i in range(len(palavra)):
    for j in range(len(palavra2)):
      if(vetorPalavra[i] == vetorPalavra2[j]):
        vetorPalavra[i] = 0
        vetorPalavra2[j] = 0
                      
                 
           
        
  for i in range(len(palavra)):
    if(vetorPalavra[i] != 0 or vetorPalavra2[i] != 0):
      permuta = False
            
      
print("A primeira palavra é uma permutação da segunda? : "+str(permuta))